### PR TITLE
Convert non-scalar context values to string representations

### PR DESCRIPTION
**Pull request by chrisminett**
As a solution to #50 and extension of #38, this converts non-scalar (and boolean) context values into PHP string representations.

### CODE REVIEW

**Comment by chrisminett** [comment]
Your comment on https://github.com/bzikarsky/gelf-php/pull/38#issuecomment-149809824 suggests using JSON notation and outputting simply 'object' for more complex types, but this implementation uses PHP-style representation and deals with complex objects too, as much as possible.

I did it this way as PHP-style seemed appropriate for a PHP project, however when considering the log output when viewed in Graylog, perhaps it would be better to have something more generic like JSON. What do you think?

File: src/Gelf/Logger.php [hunk 1]
```diff
@@ -120,10 +120,11 @@ public function setFacility($facility = null)
      * @param  array   $context
      * @return Message
      */
-    protected function initMessage($level, $message, $context)
+    protected function initMessage($level, $message, array $context)
```
Comment (bzikarsky) [inline src/Gelf/Logger.php]: :+1:

**Comment by bzikarsky** [comment]
I'm torn on your implementation. On the one hand I think it's nice to convert as many types with most of their information to a string representation. On the other hand this adds a lot of complexity.

Since `Gelf\Logger` was initially planed as a lightweight PSR-3 implementation I currently lean more towards a more simplistic conversion logic. For all the bells and whistles one can always use Monolog. It supports this library as transport. (Its `LineFormatter` https://github.com/Seldaek/monolog/blob/master/src/Monolog/Formatter/LineFormatter.php#L134-L149 seems pretty much what you are aiming for)

**Comment by chrisminett** [comment]
Yeah, discussing with my colleague here (@qubit05) we tend to agree that a simpler, more generic output would be better.

* JSON removes the PHP-specific styling in a user's logs
* Only dealing with objects that implement `__toString` and leaving others as `'object'` means that the context value will always be a true string, rather than an object representation

I'll update the PR with these changes now.

File: src/Gelf/Logger.php [hunk 2]
```diff
@@ -140,6 +141,33 @@ protected function initMessage($level, $message, $context)
...
+            switch (gettype($value)) {
+                case 'array':
+                case 'boolean':
+                    $value = json_encode($value);
+                    break;
+                case 'object':
+                    if (method_exists($value, '__toString')) {
+                        $value = (string)$value;
+                    } else {
+                        $value = '[object]';
```
Comment (bzikarsky) [inline src/Gelf/Logger.php]: Hm. One more idea: `$value = "[object (" . get_class($value) . ")]";` -- What do you think?

Comment (chrisminett) [inline src/Gelf/Logger.php]: Nice

**Comment by chrisminett** [comment]
Checking http://php.net/manual/en/function.gettype.php I've added tests and conversions for all results of `gettype()`

**Comment by bzikarsky** [comment]
Looks good - I like it! :+1: 

One final request -- Can you squash your changes to one commit with a nice commit-message? (For a definition of "nice", I'll always have http://chris.beams.io/posts/git-commit/ at the back of my mind)

**Comment by bzikarsky** [comment]
That is awesome! Thanks! :smile:

**Comment by chrisminett** [comment]
NP. Looking forward to updating our projects!

**Comment by bzikarsky** [comment]
Version-tag for 1.4.1 is in place. -- Happy coding! :)

**Pull request merged**